packages/flutter_tools/lib/src/compile.dart

```diff
@@ -314,7 +312,7 @@ class KernelCompiler {
         platformDill,
       ],
       ...?extraFrontEndOptions,
-      mainUri ?? mainPath,
```

Comment:
`mainUri` isn't possible to be null as written:
```dart
String? mainUri;
if (packagesPath != null) {
  mainUri = packageConfig.toPackageUri(mainFileUri)?.toString(); // nullable
}
mainUri ??= toMultiRootPath(mainFileUri, _fileSystemScheme, _fileSystemRoots, _fileSystem.path.separator == r'\'); // nonnull
```
I'm not sure if this is a bug and there are cases when this should be `mainPath`.  But at least how it's written `mainUri` is always used in the command.

Reply:
We added toMultiRootPath recently (relatively) so that probably changed things